### PR TITLE
Fix discrete outputmodel sampling

### LINKED ISSUE #49: discrete sampling fails during dirichlet sampling with numpy-1.14
This used to work with prior numpy version (<1.14):
```python
from pyemma import msm
dtrajs = [[0, 1, 2, 2, 2, 2, 1, 2, 2, 2, 1, 0, 0, 0, 0, 0, 0, 0],
               [0, 0, 0, 0, 1, 1, 2, 2, 2, 2, 2, 2, 2, 1, 0, 0]]  # two trajectories
mm = msm.bayesian_hidden_markov_model(dtrajs, 2, 2, show_progress=False)

```
output:
```
  File "/home/mi/marscher/workspace/pyemma/simple.py", line 4, in <module>
    mm = msm.bayesian_hidden_markov_model(dtrajs, 2, 2, show_progress=False)
  File "/home/mi/marscher/workspace/pyemma/pyemma/msm/api.py", line 1318, in bayesian_hidden_markov_model
    return bhmsm_estimator.estimate(dtrajs)
  File "/home/mi/marscher/workspace/pyemma/pyemma/_base/estimator.py", line 413, in estimate
    self._model = self._estimate(X)
  File "/home/mi/marscher/workspace/pyemma/pyemma/msm/estimators/bayesian_hmsm.py", line 288, in _estimate
    store_hidden=self.store_hidden, call_back=call_back)
  File "/home/mi/marscher/miniconda/lib/python3.6/site-packages/bhmm/api.py", line 467, in bayesian_hmm
    call_back=call_back)
  File "/home/mi/marscher/miniconda/lib/python3.6/site-packages/bhmm/estimators/bayesian_sampling.py", line 250, in sample
    self._update()
  File "/home/mi/marscher/miniconda/lib/python3.6/site-packages/bhmm/estimators/bayesian_sampling.py", line 270, in _update
    self._updateEmissionProbabilities()
  File "/home/mi/marscher/miniconda/lib/python3.6/site-packages/bhmm/estimators/bayesian_sampling.py", line 333, in _updateEmissionProbabilities
    self.model.output_model.sample(observations_by_state)
  File "/home/mi/marscher/miniconda/lib/python3.6/site-packages/bhmm/output_models/discrete.py", line 250, in sample
    self._output_probabilities[i, :] = dirichlet(self.prior[i] + count)
  File "mtrand.pyx", line 4753, in mtrand.RandomState.dirichlet
ValueError: alpha <= 0
```
Some elements of the resulting vector passed as alpha are indeed zero, but there is no insight to me, why this used to work in older numpy versions.

### PR DESCRIPTION
**Pull request by marscher**
As reported in #49, numpy's dirichlet sampling now properly checks input for elements equal to zero. This PR filters these elements by fancy indexing to allow convergence of the sampling.

It also removes the double adding of the prior, which seems unintended. 

Fixes #49.

### CODE REVIEW

**Comment by marscher** [comment]
I have also cross-tested this change with pyemma-2.5.2 and all hmm tests are still passing. @thempel, please check if you find the time (and the office is not too overheated).

**Comment by thempel** [comment]
Ok, looks fine to me (given that we were right in #49 ). Thanks for taking care of this! I think we should eventually add some new test cases, especially for the bhmm with prior.

File: bhmm/output_models/gaussian.py [hunk 1]
```diff
@@ -159,8 +159,13 @@ def _p_o(self, o):
         if self.__impl__ == self.__IMPL_C__:
             return gc.p_o(o, self.means, self.sigmas, out=None, dtype=type(o))
         elif self.__impl__ == self.__IMPL_PYTHON__:
-            C = 1.0 / (np.sqrt(2.0 * np.pi) * self.sigmas)
-            Pobs = C * np.exp(-0.5 * ((o-self.means)/self.sigmas)**2)
+            eps = 1e-14
+            if np.any(self.sigmas < eps):
+                sigmas = np.ones_like(self.sigmas)
```
Comment (thempel) [inline bhmm/output_models/gaussian.py]: You sure we should silently overwrite all Gaussian variances by one if just one element is <1e-14? That means that the estimation could get completely ignored when computing output probabilities, no? What was the intention behind this change?

Comment (thempel) [reply bhmm/output_models/gaussian.py]: The problem you were pointing to is that `_p_o` does not actually return probabilities (as stated in the first line of its docstring) but the Gaussian probability density (as stated in the last line of the docstring...). That means when evaluating close to the mean, this function has no upper boundary in the zero-variance limit (approaching the Dirac delta). 

If this causes overflows, we should rather check for this and raise an error. My guess is that the zero-variance case is not really what the Gaussian model was made for anyway.

**Comment by marscher** [comment]
applied changes as suggested by @thempel

**Review by thempel — APPROVED** [review]
Thank you very much @marscher !

**Pull request merged**